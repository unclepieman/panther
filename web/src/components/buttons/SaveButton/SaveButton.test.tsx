/**
 * Panther is a Cloud-Native SIEM for the Modern Security Team.
 * Copyright (C) 2020 Panther Labs Inc
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { render, fireEvent, waitFor } from 'test-utils';
import { Formik, Form, Field } from 'formik';
import FormikTextInput from 'Components/fields/TextInput';
import SaveButton from './index';

const TestForm = ({ onSubmit, ...rest }) => (
  <Formik
    initialValues={{
      test: '',
    }}
    onSubmit={values => {
      onSubmit(values);
    }}
  >
    {() => (
      <Form>
        <Field as={FormikTextInput} label="test" name="test" id="test" />
        <SaveButton aria-label="submit button" {...rest}>
          Submit
        </SaveButton>
      </Form>
    )}
  </Formik>
);

describe('CancelButton', () => {
  it('renders', () => {
    const onSubmit = jest.fn();
    const { container } = render(<TestForm onSubmit={onSubmit} />);
    expect(container).toMatchSnapshot();
  });

  it('allows form submission', async () => {
    const onSubmit = jest.fn();
    const { findByLabelText, getByAriaLabel } = render(<TestForm onSubmit={onSubmit} />);

    const submit = getByAriaLabel('submit button');
    const input = (await findByLabelText('test')) as HTMLInputElement;

    expect(submit).toHaveAttribute('disabled');
    // Make the button available
    await waitFor(() => {
      fireEvent.change(input, {
        target: {
          value: 'test me',
        },
      });
    });

    expect(submit).not.toHaveAttribute('disabled');
    await waitFor(() => {
      fireEvent.click(submit);
    });

    expect(onSubmit).toHaveBeenCalled();
  });
});
