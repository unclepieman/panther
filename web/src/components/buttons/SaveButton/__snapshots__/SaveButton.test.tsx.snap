// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CancelButton renders 1`] = `


.panther-3 {
  min-height: 47px;
  position: relative;
  border: 1px solid;
  -webkit-transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  background-color: transparent;
  border-radius: 4px;
  border-color: #3e516b;
}

.panther-3:hover {
  border-color: #0081c5;
}

.panther-3:focus-within {
  border-color: #0081c5;
}

.panther-3:focus-within label {
  font-weight: 500;
  -webkit-transform: translate(6px, 4px) scale(0.65);
  -moz-transform: translate(6px, 4px) scale(0.65);
  -ms-transform: translate(6px, 4px) scale(0.65);
  transform: translate(6px, 4px) scale(0.65);
}

.panther-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.panther-0 {
  vertical-align: top;
  width: 100%;
  height: 100%;
  padding-left: 16px;
  padding-top: 20px;
  padding-bottom: 8px;
  position: relative;
  color: #f6f6f6;
  font-size: 0.875rem;
  font-weight: 500;
  background-color: transparent;
  border: 0;
  z-index: 2;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.panther-0::-webkit-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  font-weight: 500;
}

.panther-0::-moz-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  font-weight: 500;
}

.panther-0:-ms-input-placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  font-weight: 500;
}

.panther-0::placeholder {
  opacity: 0;
  color: #f6f6f6;
  -webkit-transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  font-weight: 500;
}



.panther-0:focus::-webkit-input-placeholder {
  opacity: 0.4;
}

.panther-0:focus::-moz-placeholder {
  opacity: 0.4;
}

.panther-0:focus:-ms-input-placeholder {
  opacity: 0.4;
}

.panther-0:focus::placeholder {
  opacity: 0.4;
}

.panther-0:-webkit-autofill {
  -webkit-box-shadow: 0 0 0 30px #192231 inset;
  -webkit-text-fill-color: #f6f6f6;
  border-radius: 4px;
}

.panther-1 {
  pointer-events: none;
  font-size: 0.875rem;
  padding-left: 16px;
  padding-right: 16px;
  color: #bdbdbd;
  top: 0;
  position: absolute;
  transform-origin: center left;
  -webkit-transform: translate(0, 14px) scale(1);
  -moz-transform: translate(0, 14px) scale(1);
  -ms-transform: translate(0, 14px) scale(1);
  transform: translate(0, 14px) scale(1);
  -webkit-transition: -webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  font-weight: 400;
}

.panther-6 {
  width: auto;
  outline: none;
  padding-left: 20px;
  padding-right: 20px;
  height: 46px;
  font-size: 0.938rem;
  -webkit-transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: border-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,background-color 100ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  border: 1px solid;
  border-color: #67b52c;
  background-color: #67b52c;
  cursor: pointer;
  color: #f6f6f6;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.panther-6:hover {
  background-color: #71bf36;
  border-color: #71bf36;
}

.panther-6:focus {
  background-color: #71bf36;
  border-color: #bcff81;
}

.panther-6:active,
.panther-6[data-active=true] {
  background-color: #5dab22;
  border-color: #5dab22;
}

.panther-5 {
  display: inline-block;
  vertical-align: sub;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
  color: currentColor;
  margin-left: -4px;
  margin-right: 8px;
}

<div>
  <header
    id="main-header"
  />
  <form
    action="#"
  >
    <div
      class="panther-4"
    >
      <div
        class="panther-3"
      >
        <div
          class="panther-2"
        >
          <input
            aria-invalid="false"
            class="panther-0"
            id="test"
            name="test"
            type="text"
            value=""
          />
          <label
            class="panther-1"
            for="test"
          >
            test
          </label>
        </div>
      </div>
    </div>
    <button
      aria-label="submit button"
      class="panther-6"
      disabled=""
      type="submit"
    >
      <svg
        class="panther-5"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M17.946 3a1.1 1.1 0 01.916.49l1.953 2.93c.12.18.185.393.185.61V19.9a1.1 1.1 0 01-1.1 1.1H4.1A1.1 1.1 0 013 19.9V4.1A1.1 1.1 0 014.1 3zm-.482 2H16v3.9a1.1 1.1 0 01-.98 1.094L14.9 10H8.1a1.1 1.1 0 01-1.094-.98L7 8.9V5H5v14h14V7.303L17.464 5zM14 5H9v3h5V5z"
        />
      </svg>
      Submit
    </button>
  </form>
  <footer
    id="footer"
  />
</div>
`;
